docs/content/preview/architecture/docdb-replication/raft.md

```diff
@@ -0,0 +1,74 @@
+1. The leader appends the batch to its Raft log and chooses a [hybrid timestamp](../../transactions/transactions-overview#hybrid-logical-clocks) for the write operation.
+1. The data is replicated via Raft to its peers.
+1. After receiving a `replication successful` callback from the Raft subsystem, the leader applies the write batch to its local DocDB.
+1. Responds with success to the user.
+1. The next update from the leader notifies followers that the entry has been committed, and the followers apply the write batch to their RocksDB instances.
+
+## Leader election
+
+When the system starts up, each node is assigned an initial election term, which is typically a non-negative integer. When a  follower does not receive a heartbeat within the election timeout period it becomes a candidate to start a new election term and attempt to become the new Leader. Each node votes for a candidate in a particular election term. The candidate that receives votes from a majority of the nodes is elected as the new leader. Whenever a new leader is elected, the election term is incremented. This ensures that each term has a unique identifier, and it helps the nodes in the system differentiate between the various terms.
+
+If a node receives a request from a leader with a stale term (i.e., a term that is lower than the node's current term), the node will reject the request and inform the leader of the newer term. This helps prevent outdated leaders from making changes to the replicated log. RAFT ensures that the replicated log remains consistent across the nodes by requiring that all log entries be associated with a specific term. This means that a node will only accept log entries that are from the current term or a previous term
+
+The Leader continues to send heartbeat messages to maintain its leadership status. If a Follower does not receive a heartbeat within the election timeout period, a new election is triggered.
+
+## Consistency guarantees
+
+RAFT ensures that the replicated log remains consistent across all the nodes in the cluster. It achieves this by enforcing a rule that the leader can only append new entries to the log, and it must ensure that those entries are consistent with the log of the majority of the nodes. If a follower's log is not consistent with the leader's log, the leader will override the follower's log with its own.
+
+## Fault tolerance
+
+RAFT is designed to be fault-tolerant, meaning that it can continue to operate even if some of the nodes in the cluster fail. As long as a majority of the nodes in the cluster are alive, the system can continue to make progress and maintain the consistency of the replicated log. If the leader fails, a new leader will be elected, and the system will continue to operate without interruption.
```

Comment:
please don't use the word cluster here.  We cannot tolerate the loss of a minority of the nodes in a cluster, only the loss of a minority of nodes in every Raft group.
availability here requires the majority of nodes are able to talk to each other
interruption is a tricky word; there is a several second hiccup when leaders failed in an unplanned way

Reply:
that is, we are okay as long as there is no raft group for which a majority of nodes are lost